module/users/helper.js

```diff
@@ -225,36 +225,50 @@ module.exports = class UsersHelper {
 
         let totalCount = 0;
         let mergedData = [];
+        let solutionTypes = [];
+        //check if type is passed or not if not passed then create an array of all types of solutions
+        if (type === "") {
+          solutionTypes = [
+            constants.common.SURVEY,
+            constants.common.IMPROVEMENT_PROJECT,
+            constants.common.OBSERVATION,
+          ];
+        } else {
+          solutionTypes = [type];
+        }
+        let autoTargetedSolutions = [];
 
         // fetching all the targted solutions in program
-        let autoTargetedSolutions =
-          await solutionsHelper.forUserRoleAndLocation(
+        for (let i = 0; i < solutionTypes.length; i++) {
+          let solutions = await solutionsHelper.forUserRoleAndLocation(
```

Comment:
@ankitpws - As discussed we should not query in loop

Reply:
Not running Query in loop